### ISSUE TITLE
Recolor issue with high max words

### ISSUE THREAD

**Issue by Peter92** [amueller/word_cloud#307]
I was testing a script I'd done with a 1000x1000 image with 500 words, and an error kept popping up.

```
  File "C:\Program Files\Python27\lib\site-packages\wordcloud\color_from_image.py", line 49, in __call__
    return "rgb(%d, %d, %d)" % tuple(color)
TypeError: %d format: a number is required, not numpy.float64
```

I dug into the file a little bit, and found that `patch.reshape(-1, 3)` from `colour_from_image.py` line 47 was returning an empty array when it is used at the bottom of the image, which then returned `[nan, nan, nan]` when used in `np.mean`. The wordcloud size and colour image size are the exact same, so that shouldn't be the issue.

[Here is an image](https://imgur.com/a/wHm9G) with some error words coloured green.

Also if it helps, here's some locations and font sizes of where the error happened:

```
(974, 852) 16
(971, 31) 14
(974, 791) 13
(979, 553) 11
(979, 669) 10
(975, 95) 10
(981, 168) 9
(981, 15) 9
(979, 719) 9
(979, 245) 9
(979, 331) 7
(983, 104) 6
(978, 630) 6
(981, 65) 6
(980, 142) 6
(983, 644) 6
(966, 800) 5
(979, 435) 5
(970, 804) 5
(983, 522) 5
(979, 289) 5
(974, 530) 5
(982, 40) 5
(967, 0) 5
(983, 784) 5
(978, 698) 4
(984, 702) 4
(985, 362) 4
(984, 825) 4
(978, 122) 4
(969, 3) 4
(984, 141) 4
(984, 405) 4
(984, 444) 4
```

**Comment by amueller** [comment]
can you provide a self-contained example, i.e. all the code and possibly data to reproduce the error?

**Comment by Peter92** [comment]
Ahh apologies, just found it was my fault. I was using `width, height = array.shape[:2]`, which was in the wrong order, it just wasn't very noticeable as the image is almost square.

**Comment by amueller** [comment]
ah, could you please open a PR with a better error message or provide code to reproduce? That seems something where a better error might help a lot.

**Comment by Peter92** [comment]
Ah alright sure, I tried to keep your styling, also found another bug when I accidentally set the resolution to 10x10 (line 410 of `wordcloud.py` with `self.layout_`, it occasionally only had 1 value and caused an error), so threw in a quick fix for that too.

The one I posted here about was simply just when the image put into `ImageColorGenerator` had smaller dimensions than the output image.

**Comment by Peter92** [comment]
Hey sorry caught me at a slightly busy time aha, I managed to fix that particular issue (think its been merged) but accidentally introduced another one in the process, where it would assume a chunk of colour is empty if it is pure black (since all the values are `0`). 

I made the other pull request with that particular fix but completely forgot to get round to updating the tests, it was failing as I'd added in a manual check for the exception earlier that was now fixed. I'll try remove that particular test and see if it passes :)

**Comment by amueller** [comment]
I think I fixed that in #391.
